Review edits to Device Management for Mbed OS

peknis01 opened the pull request:
Friday review edits and possible comments on the **Device Management for Mbed OS** chapter.

peknis01 committed:
Review edits to device-management.md

peknis01 committed:
Review edits in device-management-config.md

peknis01 committed:
Review changes to device-management-test.md

peknis01 committed:
Review changes to introduction_chip2cloud.md

peknis01 commented on docs/pelion_ready/introduction_chip2cloud.md:
```diff
@@ -4,7 +4,7 @@ You can connect Mbed OS devices to the Pelion Iot Platform and use [the full sui
 The documentation covers:
 
-1. [Using Pelion Ready boards](../mbed-os-pelion/index.html) to establish a connection to the platform.
+1. [Using Pelion Ready boards](../mbed-os-pelion/device-management-for-mbed-os.html) to establish a connection to the platform.
```
Not entirely sure where the link for the Pelion Ready boards should lead.

iriark01 replied:
Exactly there; it's just not called Pelion Ready externally

peknis01 replied:
What should we call them? Mbed enabled?

iriark01 replied:
Sorry, didn't mean "externally", meant the page title is about DM. The name here is fine, and so is the link.

peknis01 commented:
In my opinion, the structure in the device-management-test.md was a bit funny and I did some shuffling. Also, the unit symbols for bytes and kilobytes in it were not ok, I think.

Other things I did, were some link fixes and improvements (I hope) to texts here and there.

iriark01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -68,11 +68,13 @@ Edit the `mbed_app.json` file, and create a new entry under `target_overrides` w
+At this point, we recommend that you run the "connect" test suite, which verifies that the device can successfully bootstrap, register and send and receive data from Device Management with the provided configuration.
 
-If you already configured your Device Management API key and initialized your credentials as described in the [previous section](#adding-device-management-feature-to-your-application), you can compile the "Connect" tests using:
+If you already configured your Device Management API key and initialized your credentials as described in [Adding Device Management connectivity to your application](../mbed-os-pelion/device-management-for-mbed-os.html#adding-device-management-connectivity-to-your-application), you can compile the "connect" tests as follows:
```
If this is in the same page, you don't need the book and page names in the link. Just the #header

iriark01 replied:
You know I remove "as follows" wherever I see it, right?

peknis01 replied:
You are absolutely right. I already forgot it was on the same page.

peknis01 replied:
Then again, I did check that this is not actually on the same page. It is in a different doc, therefore with the relative link.

iriark01 replied:
Sorry, my bad

peknis01 replied:
- Fair enough. I'll try to remember to stop using "as follows". 
- I remove "please", whenever I see it. We don't need to be that polite in the docs.

iriark01 replied:
We're a British company. We say "please" and make our PRs queue properly.

peknis01 replied:
We are not a British company. We are Japanese, remember.

melwee01 replied:
🤣

iriark01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -84,13 +86,19 @@ To run the tests:
+## Bootloader configuration
 
-After you've successfully passed the "Connect" tests as described above, you can enable firmware update feature by adding a bootloader to your application.
+When the "connect" tests have completed successfully, you can enable the firmware update feature by adding a bootloader to your application.
```
Completing and passing tests are not the same thing

peknis01 replied:
Right again. Needs to be passed.

peknis01 committed:
completed -> passed

If a test passes, it is a success, therefore, leaving out "successfully".

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -2,47 +2,47 @@
 Device Management configuration has five distinct areas:
 
-- Connectivity - the transport type for the device to connect to the device management service.
+- Connectivity - the transport type for the device to connect to the Device Management service.
```
: instead of - ?

peknis01 replied:
Already one : above, so I decided to leave this as is.

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -2,47 +2,47 @@
+- Connectivity - the transport type for the device to connect to the Device Management service.
 - Storage - the storage type and writing used for both the credentials and the firmware storage.
-- Flash geometry - the device flash "sandbox" for bootloader, application header, application image and two SOTP regions.
+- Flash geometry - the device flash "sandbox" for bootloader, application header, application image and two Software One Time Programming (SOTP) regions.
```
One-Time (capitalization?)

peknis01 replied:
We have this without in our glossary. However, the glossary may be wrong.

melwee01 replied:
@AnotherButler, this one's for you.

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -2,47 +2,47 @@
 Except for connectivity, the majority of the configuration is shared between the application and bootloader, which ensures the bootloader can correctly find, verify, authorize and apply an update to the device application.
 
-For full documentation about bootloaders and firmware update, please read the following documents:
+For more details about bootloaders and firmware update, read the following documents:
```
```suggestion
For more information on bootloaders and firmware update, see:
```

AnotherButler replied:
When we send people elsewhere, we should use "please".

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -2,47 +2,47 @@
+   Valid options are `SD`, `SPIF`, `QSPIF` and `FLASHIAP` (not recommended). For more available options, see the [block device components](https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice).
 
-   You also have to specify block device pin configuration, which may vary from one block device type to another. Here's an example for `SD`:
+   You also have to specify the block device pin configuration, which may vary from one block device type to another. Here's an example for `SD`:
```
```suggestion
   You must also specify the block device pin configuration, which may vary from one block device type to another. Here's an example for `SD`:
```

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -84,13 +86,19 @@ To run the tests:
+## Bootloader configuration
 
-After you've successfully passed the "Connect" tests as described above, you can enable firmware update feature by adding a bootloader to your application.
+When the "connect" tests have passed, you can enable the firmware update feature by adding a bootloader to your application.
```
Are there any circumstances under which someone wouldn't do this? Should it just be an imperative 'enable' minus 'you can'?

melwee01 commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -132,9 +140,9 @@ After you've successfully passed the "Connect" tests as described above, you can
 
 <span class="notes">**Note:** `mbed-bootloader` is primarily optimized for `GCC_ARM`, so you may want to compile it with that toolchain.
-Before jumping to the next step, you should compile and flash the bootloader and then connect over the virtual comport to ensure the bootloader is running correctly. You can ignore errors related to checksum verification or failure to jump to application - these are expected at this stage.</span>
+Before adding the bootloader to your application, you need to compile and flash the bootloader and then connect over the virtual com port to ensure the bootloader is running correctly. You can ignore errors related to checksum verification or failure to jump to application - these are expected at this stage.</span>
```
'You must', or just 'compile'.

melwee01 commented on docs/pelion_ready/device-management-test.md:
````diff
@@ -55,31 +12,32 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
     ```
 
-    For instructions on how to generate an API key, please [see the documentation](https://cloud.mbed.com/docs/current/integrate-web-app/api-keys.html#generating-an-api-key).
+    See [instructions on how to generate an API key](https://www.pelion.com/docs/device-management/current/integrate-web-app/api-keys.html#generating-an-api-key).
````
Should we change links in this, or do it all at once in its own PR?

peknis01 replied:
Didn't I just change them? I may have missed some.

melwee01 commented on docs/pelion_ready/device-management-test.md:
```diff
@@ -55,31 +12,32 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
+    This creates your private and public key pair and also initializes various `.c` files with these credentials, so that you can use Device Management Connect and (firmware) Update.
 
-1. Remove the `main.cpp` application from the project, or ensure the content of the file is wrapped with `#ifndef MBED_TEST_MODE`.
+1. Remove the `main.cpp` application from the project, or ensure that the content of the file is wrapped with `#ifndef MBED_TEST_MODE`.
```
Thought we didn't use 'wrapped' or 'wrapper'.

AnotherButler replied:
This "that" is unnecessary.

melwee01 commented on docs/pelion_ready/device-management-test.md:
```diff
@@ -55,31 +12,32 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
-1. Remove the `main.cpp` application from the project, or ensure the content of the file is wrapped with `#ifndef MBED_TEST_MODE`.
+1. Remove the `main.cpp` application from the project, or ensure that the content of the file is wrapped with `#ifndef MBED_TEST_MODE`.
 
 1. Compile the tests with the `MBED_TEST_MODE` compilation flag.
```
```suggestion
1. Compile the tests with the `MBED_TEST_MODE` compilation flag:
```

melwee01 commented on docs/pelion_ready/device-management-test.md:
Depending on what?

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -1,8 +1,8 @@
 <h1 id="device-management">Device Management for Mbed OS</h1>
 
-Pelion Client is an abstraction of the Device Management Client, and lets Mbed OS devices use the [Pelion Device Management Services](https://cloud.mbed.com/docs/latest/welcome/index.html). Pelion Client:
+Pelion Client is an abstraction of Device Management Client, and lets Mbed OS devices use the [Pelion Device Management services](https://www.pelion.com/docs/device-management/current/welcome/index.html). Pelion Client:
```
```suggestion
Pelion Client is an abstraction of Device Management Client, and lets Mbed OS devices use [Pelion Device Management services](https://www.pelion.com/docs/device-management/current/welcome/index.html). Pelion Client:
```

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -1,8 +1,8 @@
+Pelion Client is an abstraction of Device Management Client, and lets Mbed OS devices use the [Pelion Device Management services](https://www.pelion.com/docs/device-management/current/welcome/index.html). Pelion Client:
 
-- Provides LwM2M resources - variables that sync automatically through Device Management. Pelion Client makes it trivial to expose sensors, actuators and other variables to a cloud service.
+- Provides LwM2M Resources - variables that sync automatically through Device Management. Pelion Client makes it trivial to expose sensors, actuators and other variables to a cloud service.
```
```suggestion
- Provides LwM2M Resources, which are variables that sync automatically through Device Management. Pelion Client makes it possible to expose sensors, actuators and other variables to a cloud service.
```
Expose?

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
 ### Requirements
 
-Not every device (microcontroller, module or board) is capable of running device management features. Although you can add or extend some hardware capabilities, such as connectivity, storage and TRNG, others are impossible or inconvenient to extend (for example, RAM or flash).
+Not every device (microcontroller, module or board) is capable of running device management features. Although you can add or extend some hardware capabilities, such as connectivity, storage and TRNG. Others are impossible or inconvenient to extend (for example, RAM or flash).
```
Device management or Device Management? Are we talking general concepts or our product in particular?

melwee01 replied:
```suggestion
Not every device (microcontroller, module or board) is capable of running device management features. You can add or extend some hardware capabilities, such as connectivity, storage and TRNG. Others are impossible or inconvenient to extend (for example, RAM or flash).
```

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
 
-- RAM: 128K or more.
-- Flash: 512K or more.
+- RAM: 128 K or more.
```
KB, no?

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
-- RAM: 128K or more.
-- Flash: 512K or more.
+- RAM: 128 K or more.
+- Flash: 512 K or more.
```
Same as above.

melwee01 commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
+- RAM: 128 K or more.
+- Flash: 512 K or more.
 - Real Time Clock (RTC).
 - (Optional but recommended) True Random Number Generator (TRNG).
```
```suggestion
-  True Random Number Generator (TRNG). Optional, but recommended.
```

AnotherButler replied:
Why rearrange this one?

peknis01 replied:
Either is ok for me.

AnotherButler commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -2,47 +2,47 @@
    The possible options are `ETHERNET`, `WIFI` and `CELLULAR`.
 
-   Depending on connectivity type, you might have to specify more config options. For an example, please see the defined `CELLULAR` targets in `mbed_app.json`.
+   Depending on connectivity type, you may have to specify more options. For an example, see the defined `CELLULAR` targets in `mbed_app.json`.
```
Please see earlier comment about "please".

peknis01 replied:
Not willing to be that polite.

AnotherButler commented on docs/pelion_ready/device-management-config.md:
````diff
@@ -2,47 +2,47 @@
    ```
 
-   Valid options are `SD`, `SPIF`, `QSPIF` and `FLASHIAP` (not recommended). For more available options, please see the [block device components](https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice).
+   Valid options are `SD`, `SPIF`, `QSPIF` and `FLASHIAP` (not recommended). For more available options, see the [block device components](https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice).
````
Please see earlier comment about "please".

peknis01 replied:
Still not that polite.

AnotherButler commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -68,11 +68,13 @@ Edit the `mbed_app.json` file, and create a new entry under `target_overrides` w
+   - `sotp-num-sections` should always be set to `2`.
 
-At this point, we recommend you run the "connect" test suite, which verifies that the device can successfully bootstrap, register and send and receive data from Device Management with the provided configuration.
+At this point, we recommend that you run the "connect" test suite, which verifies that the device can successfully bootstrap, register and send and receive data from Device Management with the provided configuration.
```
This "that" is unnecessary.

AnotherButler commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -84,13 +86,19 @@ To run the tests:
+  
+    `mbed import mbed-bootloader-extended`.
+
+1. In the imported bootloader application, edit the configuration, for example `mbed-bootloader-extended/mbed_app.json`. Add a new target entry and specify:
```
"entry, and"

peknis01 replied:
Academic comma dilemma. I'd rather not have that.

iriark01 replied:
I don't think it's academic; +1 for adding it.

AnotherButler commented on docs/pelion_ready/device-management-config.md:
```diff
@@ -132,9 +140,9 @@ After you've successfully passed the "Connect" tests as described above, you can
+Before adding the bootloader to your application, you need to compile and flash the bootloader and then connect over the virtual com port to ensure the bootloader is running correctly. You can ignore errors related to checksum verification or failure to jump to application - these are expected at this stage.</span>
 
-## 3. Add the bootloader to your application
+## Adding the bootloader to your application
```
Do these steps need to happen in order? If so, is there a way to indicate that without numbering the headings?

peknis01 replied:
Numbering in headings is a bit questionable.

melwee01 replied:
Heading levels, perhaps, but less noticeable in the rendered doc than the raw markdown.

iriark01 replied:
Manual numbering goes wrong very quickly. I've asked for automated numbering to be added to the CSS, but we don't have it yet.

AnotherButler commented on docs/pelion_ready/device-management-test.md:
````diff
@@ -55,31 +12,32 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
     ```
 
-    This creates your private and public key pair and also initializes various `.c` files with these credentials, so you can use Device Management Connect and (firmware) Update.
+    This creates your private and public key pair and also initializes various `.c` files with these credentials, so that you can use Device Management Connect and (firmware) Update.
````
This "that" is unnecessary.

AnotherButler commented on docs/pelion_ready/device-management-test.md:
```diff
@@ -93,23 +51,66 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
 ## Troubleshooting
 
-Below are common issues and fixes.
+Common issues and fixes to them.
```
It bothers me that this doesn't have a verb. Can we delete this line entirely?

AnotherButler commented on docs/pelion_ready/device-management-test.md:
```diff
@@ -93,23 +51,66 @@ Device Management tests rely on the Python SDK to test the end-to-end solution.
 ### Device identity is inconsistent
 
-If your device ID in Device Management is inconsistent when your device resets, it could be because it is failing to open the credentials on the storage held in the Enhanced Secure File System. Typically, this is because the device cannot access the Root of Trust stored in SOTP.
+If your device ID in Device Management is inconsistent when your device resets, the device is probably failing to open the credentials on the storage held in the Enhanced Secure File System. Typically, the reason is that the device cannot access the Root of Trust stored in SOTP.
```
This "that" is unnecessary.

AnotherButler commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
 - IP connectivity: Ethernet, Wi-Fi, cellular, 6LoWPAN or Thread.
 
-Additionally, we recommend the latest version of Mbed OS support the device and any additional complementary hardware components, or that they have support for the APIs provided in the latest releases of Mbed OS.
+Additionally, we recommend that the latest version of Mbed OS supports the device and any additional complementary hardware components, or that they have support for the APIs provided in the latest releases of Mbed OS.
```
This "that" is unnecessary.
Query: How formal are we when it comes to the subjunctive mood? Do we keep it as "support," or do we relax it?

AnotherButler commented on docs/pelion_ready/device-management.md:
```diff
@@ -17,30 +17,30 @@ This guide builds on the [quick connect guide](https://os.mbed.com/guides/connec
-- Check the [storage options available](../reference/storage.html).
-- Check the [network options available](../reference/networking.html).
+- [Supported Mbed OS platforms](https://cloud.mbed.com/quick-start).
+- [Storage options available](../reference/storage.html).
```
Could we get rid of "available" here and just have "Storage options" and "Network options"?

melwee01 replied:
Yes.

peknis01 replied:
Did that already.

melwee01 replied:
Ah, whoops. Sorry!

AnotherButler reviewed (commented):
This looks SO much better 👍 

I've left some comments and queries for y'all.

Commit:
Update docs/pelion_ready/device-management-config.md

Co-Authored-By: peknis01 <pekka.niskanen@arm.com>

Commit:
Update docs/pelion_ready/device-management-config.md

Co-Authored-By: peknis01 <pekka.niskanen@arm.com>

Commit:
Update docs/pelion_ready/device-management.md

Co-Authored-By: peknis01 <pekka.niskanen@arm.com>

Commit:
Update docs/pelion_ready/device-management.md

Co-Authored-By: peknis01 <pekka.niskanen@arm.com>

Commit:
Update docs/pelion_ready/device-management.md

Co-Authored-By: peknis01 <pekka.niskanen@arm.com>

peknis01 committed:
Review fixes in device-management.md

peknis01 committed:
K -> KiB in device-management.md

peknis01 committed:
Review fix in device-management-config.md

peknis01 committed:
More review fixes in device-management-config.md

peknis01 committed:
Review edits in device-management-test.md

Commit:
Remove extra word in device-management.md

Remove extra word as suggested in comments.

Commit:
Add comma in device-management-config.md

Add comma before coordinating conjunction separating two independent clauses as suggested in comments.

peknis01 commented:
I'm tempted to click the Merge button. Any objections?

AnotherButler commented:
None from me
